src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2203,6 +2203,57 @@ Node TheoryStringsRewriter::rewriteReplace( Node node ) {
       return returnRewrite(node, res, "repl-subst-idx");
     }
   }
+
+  if (node[0].getKind() == STRING_STRREPL)
+  {
+    Node x = node[0];
+    Node y = node[1];
+    Node z = node[2];
+    if (x[0] == x[2] && x[0] == y)
+    {
+      // (str.replace (str.replace y w y) y z) -->
```

Comment:
I have some hesitations about this current rewrite in terms of termination. 
I'm wondering here if both of these rewrite to `(str.replace y w z)`?
It seems that the side conditions ensure both allow us to simplify to a single replace, which would be great.

Reply:
Yeah, they unfortunately don't quite rewrite to `(str.replace y w z)`. My intuition for why the right-hand side is slightly better is because you have at most one of the two replaces doing anything. On the left-hand side, you may have both replaces that change the string. It is maybe not the strongest argument but that's the best reason that I could come up with when deciding on an implementation.

Reply:
Ok, I think this is fine for now.